src/odemis/acq/stream/_projection.py

```diff
@@ -768,12 +768,15 @@ def _computeSpec(self):
         raw_md = self.stream.calibrated.value.metadata
         md = raw_md.copy()
         md[model.MD_DIMS] = "XC"  # RGB format
-        md[MD_PIXEL_SIZE] = (None, pxs)  # for the spectrum, use get_spectrum_range()
+        # Calculate the x pixel size.
+        wlstep = (self.stream._wl_px_values[-1] - self.stream._wl_px_values[0]) / len(self.stream._wl_px_values)
```

Comment:
I think you are on the right track to find out what's wrong (ie, something with the PIXEL_SIZE metadata), but that's not the proper way to fix it. The current code works fine in the GUI, and it does the right thing (because there is no "size" on the C dimension). So it's the `line_to_export_data()` which must be fixed to handle the data as expected.